src/main/scala/com/webtrends/harness/component/akkahttp/client/SimpleHttpClient.scala

```diff
@@ -0,0 +1,46 @@
+package com.webtrends.harness.component.akkahttp.client
+
+import akka.actor.Actor
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest, HttpResponse, StatusCodes}
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.webtrends.harness.logging.ActorLoggingAdapter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SimpleHttpClient extends ActorLoggingAdapter { this: Actor =>
+
+  implicit val executionContext: ExecutionContext
+  implicit val materializer = ActorMaterializer()
+
+  private val http = Http(context.system)
+
+  def request(req: HttpRequest): Future[(HttpResponse, Array[Byte])] = {
+    http.singleRequest(req).flatMap { response =>
+      response.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(bs => (response, bs.toArray))
+    }
+  }
+
+  def requestAs[T](req: HttpRequest, mapper: Array[Byte] => T): Future[(HttpResponse, T)] = {
+    request(req).map { case (response, bytes) =>
+      (response, mapper(bytes))
```

Comment:
Akka http has unmarshallers. Why not pass an instance of an Unmarshaller instead of a mapper?
https://doc.akka.io/docs/akka-http/current/common/unmarshalling.html